File: Tang/Tang/Content/General/Categories/FoundationCategory/UIDevice+Ext.m
//
//  UIDevice+Ext.m
//
//  Created by WangBo (developforapple@163.com) on 2017/4/13.
//  Copyright © 2017年 WangBo. All rights reserved.
//

#import "UIDevice+Ext.h"
#include <sys/utsname.h>

@implementation UIDevice (Ext)

+ (NSString *)hardwareName
{
    static NSString *hardware;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        struct utsname sysInfo;
        uname(&sysInfo);
        hardware = [NSString stringWithUTF8String:sysInfo.machine];
    });
    return hardware;
}

+ (BOOL)hardwareIsSimulator
{
    NSString *name = [self hardwareName];
    return [name isEqualToString:@"x86_64"] || [name isEqualToString:@"i386"];
}

@end

File: Tang/Tang/General/Extension/UIKit/UIDevice+Ext.swift
//
//  UIDevice+Ext.swift
//  Laidian
//
//  Created by Jay on 2018/1/31.
//  Copyright © 2018年 来电科技. All rights reserved.
//

import Foundation
import UIKit

extension UIDevice {
    
    static var hardwareName : String = {
        var info = utsname.init()
        uname(&info)
        let machineMirror = Mirror.init(reflecting: info.machine)
        let machine = machineMirror.children.reduce("") { (string, item) -> String in
            
            if let v = item.value as? Int8, v != 0 {
                let s = String.init(UnicodeScalar.init(UInt8(v)))
                return string + s
            }
            return string
        }
        return machine
    }()
    
    static var hardwareIsSimulator : Bool = {
        return UIDevice.hardwareName == "x86_64" || UIDevice.hardwareName == "i386"
    }()
}
